hal/network/ncp_client/quectel/quectel_ncp_client.cpp

```diff
@@ -1186,6 +1185,9 @@ int QuectelNcpClient::configureApn(const CellularNetworkConfig& conf) {
         auto lenCcid = getIccidImpl(buf, sizeof(buf));
         CHECK_TRUE(lenCcid > 5, SYSTEM_ERROR_AT_NOT_OK);
         netConf_ = networkConfigForIccid(buf, lenCcid);
+        if (netConf_.hasNetProv()) {
+            netProv_ = netConf_.netProv();
```

Comment:
Nit: It should probably be fine not to have a separate member variable and just access it through `netConf_.netProv()`, assuming we properly patch it in case of stored credentials.

Same with SARA code.

For convenience one could add a helper method to access provider directly somewhat similar to `ncpId()` for example.

Reply:
I'm not sure what you mean here by "assuming we properly patch it in case of stored credentials".  AFAIK, this extra piece of `netProv` info is not going to or need to be saved to flash memory.

For now, I don't think we need more convenience since it's just one check for `if Twilio`.

Reply:
> assuming we properly patch it in case of stored credentials

https://github.com/particle-iot/device-os/pull/2185#discussion_r485960534

I on the other hand am not sure what you mean by saving to flash memory :) `netConf_` is completely local to the NCP Client class, even if it gets reloaded in `connect()` by the cellular network manager.